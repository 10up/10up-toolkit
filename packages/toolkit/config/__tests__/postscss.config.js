const postcss = require('postcss');
const fs = require('fs');
const path = require('path');
const configFactory = require('../postcss.config');
const { loadPostCSSPlugins } = require('../../test-utils/load-postcss-plugins');

describe('postcss.config.js', () => {
	it('builds a proper postcss config file for editor styles', () => {
		expect(
			configFactory({
				file: '/path/to/editor-style.css',
				env: 'production',
			}),
		).toMatchSnapshot('env=production');

		expect(
			configFactory({
				file: '/path/to/editor-style.css',
				env: 'development',
			}),
		).toMatchSnapshot('env=development');
	});

	it('build a proper postcss config file for non-editor styles', () => {
		expect(
			configFactory({
				file: '/path/to/regular.css',
				env: 'production',
			}),
		).toMatchSnapshot('env=production');

		expect(
			configFactory({
				file: '/path/to/regular.css',
				env: 'development',
			}),
		).toMatchSnapshot('env=development');
	});
});

describe('postcss', () => {
	const config = configFactory({
		// this is just to control the postcss config based on file name
		file: 'style.css',
		env: 'development',
	});

	const postCSSPlugins = loadPostCSSPlugins(config);

	it('properly transforms css', async () => {
		const css = fs.readFileSync(path.join(__dirname, '__fixtures__', 'styleguide.css'));
		const result = await postcss(postCSSPlugins).process(css, {
			from: 'styleguide.css',
			to: 'styleguide.out.css',
		});

		expect(result.css).toMatchSnapshot();
	});

	it('transforms mixins properly', async () => {
		const css = fs.readFileSync(path.join(__dirname, '__fixtures__', 'mixins.css'));

		const result = await postcss(postCSSPlugins).process(css, {
			from: 'mixins.css',
			to: 'mixins.out.css',
		});

		expect(result.css).toMatchSnapshot();
	});

	it('transforms accordion.css properly', async () => {
		const css = fs.readFileSync(path.join(__dirname, '__fixtures__', 'accordion.css'));

		const result = await postcss(postCSSPlugins).process(css, {
			from: 'accordion.css',
			to: 'accordion.out.css',
		});

		expect(result.css).toMatchSnapshot();
	});

	it('transforms nesting.css properly', async () => {
		const css = fs.readFileSync(path.join(__dirname, '__fixtures__', 'nesting.css'));

		const result = await postcss(postCSSPlugins).process(css, {
			from: 'nesting.css',
			to: 'nesting.out.css',
		});

		expect(result.css).toMatchSnapshot();
	});
});
