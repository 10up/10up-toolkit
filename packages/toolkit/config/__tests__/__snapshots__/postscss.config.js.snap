// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postcss.config.js build a proper postcss config file for non-editor styles: env=development 1`] = `
Object {
  "plugins": Object {
    "cssnano": false,
    "postcss-import": Object {},
    "postcss-mixins": Object {},
    "postcss-object-fit-images": Object {},
    "postcss-preset-env": Object {
      "autoprefixer": Object {
        "grid": true,
      },
      "stage": 0,
    },
  },
}
`;

exports[`postcss.config.js build a proper postcss config file for non-editor styles: env=production 1`] = `
Object {
  "plugins": Object {
    "cssnano": Object {
      "preset": Array [
        "default",
        Object {
          "autoprefixer": false,
          "calc": Object {
            "precision": 8,
          },
          "convertValues": true,
          "discardComments": Object {
            "removeAll": true,
          },
          "mergeLonghand": false,
          "zindex": false,
        },
      ],
    },
    "postcss-import": Object {},
    "postcss-mixins": Object {},
    "postcss-object-fit-images": Object {},
    "postcss-preset-env": Object {
      "autoprefixer": Object {
        "grid": true,
      },
      "stage": 0,
    },
  },
}
`;

exports[`postcss.config.js builds a proper postcss config file for editor styles: env=development 1`] = `
Object {
  "plugins": Object {
    "cssnano": false,
    "postcss-editor-styles": Object {
      "ignore": Array [
        ":root",
        ".edit-post-visual-editor.editor-styles-wrapper",
        ".wp-toolbar",
      ],
      "remove": Array [
        "html",
        ":disabled",
        "[readonly]",
        "[disabled]",
      ],
      "scopeTo": ".editor-styles-wrapper",
      "tags": Array [
        "button",
        "input",
        "label",
        "select",
        "textarea",
        "form",
      ],
    },
    "postcss-import": Object {},
    "postcss-mixins": Object {},
    "postcss-object-fit-images": Object {},
    "postcss-preset-env": Object {
      "autoprefixer": Object {
        "grid": true,
      },
      "stage": 0,
    },
  },
}
`;

exports[`postcss.config.js builds a proper postcss config file for editor styles: env=production 1`] = `
Object {
  "plugins": Object {
    "cssnano": Object {
      "preset": Array [
        "default",
        Object {
          "autoprefixer": false,
          "calc": Object {
            "precision": 8,
          },
          "convertValues": true,
          "discardComments": Object {
            "removeAll": true,
          },
          "mergeLonghand": false,
          "zindex": false,
        },
      ],
    },
    "postcss-editor-styles": Object {
      "ignore": Array [
        ":root",
        ".edit-post-visual-editor.editor-styles-wrapper",
        ".wp-toolbar",
      ],
      "remove": Array [
        "html",
        ":disabled",
        "[readonly]",
        "[disabled]",
      ],
      "scopeTo": ".editor-styles-wrapper",
      "tags": Array [
        "button",
        "input",
        "label",
        "select",
        "textarea",
        "form",
      ],
    },
    "postcss-import": Object {},
    "postcss-mixins": Object {},
    "postcss-object-fit-images": Object {},
    "postcss-preset-env": Object {
      "autoprefixer": Object {
        "grid": true,
      },
      "stage": 0,
    },
  },
}
`;
